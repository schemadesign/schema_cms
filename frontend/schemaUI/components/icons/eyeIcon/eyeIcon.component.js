import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import EyeSvg from '../../../images/icons/eye.svg';
import { getStyles } from './eyeIcon.styles';
import { withStyles } from '../../styles/withStyles';

export class EyeIconComponent extends PureComponent {
  static propTypes = {
    customStyles: PropTypes.object,
  };

  render() {
    const { customStyles, theme, ...restProps } = this.props;
    const containerStyles = getStyles(theme);
    const styles = { ...containerStyles, ...customStyles };

    return <EyeSvg {...restProps} style={styles} />;
  }
}

export const EyeIcon = withStyles(EyeIconComponent);
