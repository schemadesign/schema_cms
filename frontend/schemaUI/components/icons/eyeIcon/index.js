export { EyeIcon } from './eyeIcon.component';
