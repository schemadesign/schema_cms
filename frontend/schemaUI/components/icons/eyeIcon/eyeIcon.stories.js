import React from 'react';
import { storiesOf } from '@storybook/react';

import { EyeIcon } from './eyeIcon.component';
import { withTheme } from '../../../.storybook/decorators';

storiesOf('Icons/EyeIcon', module).add('Default', () => <EyeIcon />);

const defaultProps = { customStyles: { stroke: 'blue' } };
storiesOf('Icons/EyeIcon', module)
  .addDecorator(withTheme())
  .add('with custom styles', () => <EyeIcon {...defaultProps} />);
