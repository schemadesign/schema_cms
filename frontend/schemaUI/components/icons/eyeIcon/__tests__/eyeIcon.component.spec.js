import React from 'react';
import { shallow } from 'enzyme';

import { EyeIconComponent } from '../eyeIcon.component';

describe('EyeIcon: Component', () => {
  const defaultProps = {};

  const component = props => <EyeIconComponent {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render with custom styles', () => {
    const customStyles = {
      fill: 'blue',
    };

    const wrapper = render({ customStyles });
    global.expect(wrapper).toMatchSnapshot();
  });
});
