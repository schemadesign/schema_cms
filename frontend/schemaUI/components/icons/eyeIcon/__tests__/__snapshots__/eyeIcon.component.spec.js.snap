// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EyeIcon: Component should render correctly 1`] = `
<eye.svg
  style={
    Object {
      "height": "30px",
      "stroke": "#FFF",
      "width": "30px",
    }
  }
/>
`;

exports[`EyeIcon: Component should render with custom styles 1`] = `
<eye.svg
  style={
    Object {
      "fill": "blue",
      "height": "30px",
      "stroke": "#FFF",
      "width": "30px",
    }
  }
/>
`;
